src/common/proc_macro/src/estimate_size.rs

```diff
@@ -0,0 +1,150 @@
+
+    false
+}
+
+pub fn has_nested_flag_attribute_list(
+    list: &[syn::Attribute],
+    name: &'static str,
+    flag: &'static str,
+) -> bool {
+    for attr in list.iter() {
+        if has_nested_flag_attribute(attr, name, flag) {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn extract_ignored_generics_list(list: &[syn::Attribute]) -> Vec<String> {
+    let mut collection = Vec::new();
+
+    for attr in list.iter() {
+        let mut list = extract_ignored_generics(attr);
+
+        collection.append(&mut list);
+    }
+
+    collection
+}
+
+pub fn extract_ignored_generics(attr: &syn::Attribute) -> Vec<String> {
+    let mut collection = Vec::new();
+
+    if let Ok(meta) = attr.parse_meta() {
+        if let Some(ident) = meta.path().get_ident() {
+            if &ident.to_string() != "get_size" {
+                return collection;
+            }
+            if let syn::Meta::List(list) = meta {
+                for nested in list.nested.iter() {
+                    if let syn::NestedMeta::Meta(nmeta) = nested {
+                        let ident = nmeta
+                            .path()
+                            .get_ident()
+                            .expect("Invalid attribute syntax! (no iden)");
+                        if &ident.to_string() != "ignore" {
+                            panic!(
+                                "Invalid attribute syntax! Unknown name {:?}",
+                                ident.to_string()
+                            );
+                        }
+
+                        if let syn::Meta::List(list) = nmeta {
+                            for nested in list.nested.iter() {
+                                if let syn::NestedMeta::Meta(syn::Meta::Path(path)) = nested {
+                                    let path = path
+                                        .get_ident()
+                                        .expect("Invalid attribute syntax! (no ident)")
+                                        .to_string();
+                                    collection.push(path);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    collection
+}
+
+// Add a bound `T: EstimateSize` to every type parameter T, unless we ignore it.
+pub fn add_trait_bounds(mut generics: syn::Generics, ignored: &[String]) -> syn::Generics {
+    for param in &mut generics.params {
+        if let syn::GenericParam::Type(type_param) = param {
+            let name = type_param.ident.to_string();
+            let mut found = false;
+            for ignored in ignored.iter() {
+                if ignored == &name {
+                    found = true;
+                    break;
+                }
+            }
+            if found {
+                continue;
+            }
+            type_param.bounds.push(syn::parse_quote!(EstimateSize));
+        }
+    }
+    generics
+}
```

Comment:
Can we also port tests?

Reply:
It simply won't compile if there is a bug in proc macro